packages/grafana-data/src/geo/layer.ts

```diff
@@ -0,0 +1,65 @@
+import { RegistryItemWithOptions } from '../utils/Registry';
+import BaseLayer from 'ol/layer/Base';
+import Map from 'ol/Map';
+import { PanelData } from '../types';
+import { GrafanaTheme2 } from '../themes';
+import { PanelOptionsEditorBuilder } from '../utils';
+import { ReactNode } from 'react';
+
+/**
+ * This gets saved in panel json
+ *
+ * depending on the type, it may have additional config
+ *
+ * This exists in `grafana/data` so the types are well known and extendable but the
+ * layout/frame is control by the map panel
+ *
+ * @alpha
+ */
+export interface MapLayerConfig<TCustom = any> {
+  type: string;
+  name?: string; // configured display name
+
+  // Layer transparency
+  transparency?: number;
```

Comment:
We could leave this out now and add it when it is used -- the point is to make sure we have a level of options that apply to all layers regardless of type.  Other values will include:
https://openlayers.org/en/latest/apidoc/module-ol_layer_Base-BaseLayer.html